op-chain-ops/script/script.go

```diff
@@ -365,6 +397,47 @@ func (h *Host) HasPrecompileOverride(addr common.Address) bool {
 	return ok
 }
 
+// onEnter is a trace-hook, which we use to apply changes to the state-DB, to simulate isolated broadcast calls,
+// for better gas estimation of the exact broadcast call execution.
+func (h *Host) onEnter(depth int, typ byte, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int) {
+	if len(h.callStack) == 0 {
+		return
+	}
+	parentCallFrame := h.callStack[len(h.callStack)-1]
+	if parentCallFrame.Prank == nil {
+		return
+	}
+
+	if parentCallFrame.Prank.Broadcast {
+		if vm.OpCode(typ) == vm.CALL && to != VMAddr {
+			sender := parentCallFrame.Ctx.Caller()
+			if parentCallFrame.Prank.Sender != nil {
+				sender = *parentCallFrame.Prank.Sender
+			}
+
+			h.state.SetNonce(
+				sender,
+				h.state.GetNonce(sender)+1,
+			)
```

Comment:
To be entirely correct, although not applicable to the deployment txs themselves, we should wind back the nonce of the `parentCallFrame.Ctx.Address`, if a prank was active, and if it was  a `vm.CREATE`.

Reply:
I updated this PR with nonce logging enabled. It looks to me like all the nonces are correct. The nonce of `address(this)` in the deploy script increments from 1 to 3, which is what we expect given that we are both performing a regular `call()` and a `CREATE`. You can see this on line 136 in `ScriptExample.s.sol`.

I implemented the nonce unwinding code you described here, but this reset the nonce of the deploy script to zero, which I believe is incorrect. Please let me know if I'm missing something here - I think I followed all your advice on Discord but was unable to reproduce the nonce issue you described.

Reply:
I stand corrected, the caller override is applied before the nonce-bump for Create(), so there is no need to do anything with the original caller nonce in the Create case.